### PR TITLE
Use IAM authentication for the database.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've added in a new class `RDSCredentialsProvider`

This is not a Keycloak SPI, it's an extension of the Quarkus server that
Keycloak runs on so it doesn't follow the usual SPI structure. This is
why it needs its own quarkus.properties file

https://quarkus.io/guides/credentials-provider#custom-credentials-provider

The rest is adding the jar to the Docker image and some new
dependencies.

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Would you be able to add a section in the `README` about this, similar to the SPI sections that are already there.

**Comment by MancunianSam** [comment]
I've added in a section to the README and corrected a couple of bits in other sections. I've also deleted the Jenkinsfiles.

File: README.md [hunk 1]
```diff
@@ -91,14 +91,21 @@ The following events trigger the publishing service:
...
+There is a cache which refreshes every 5 minutes so we're not calling AWS for every new database connection.
 
 ## Updating TDR Realm Configuration json
 
 A separate GitHub actions workflow is used to update the TDR realm configuration: [TDR Update Auth Server Configuration](https://github.com/nationalarchives/tdr-auth-server/actions/workflows/update.yml)
 
 This is because on container start up, if a realm already exists, its configuration json is ignored. So restarting the container will not update the configuration.
 
-The TDR Auth Server Update Jenkins job updates existing realm configuration using the Keycloak REST APIs: https://www.keycloak.org/docs-api/11.0/rest-api/index.html
+The TDR Update Auth Server Configuration GitHub actions workflow updates existing realm configuration using the Keycloak REST APIs: https://www.keycloak.org/docs-api/11.0/rest-api/index.html
```
Comment (TomJKing) [inline README.md:108]: The Keycloak API docs link needs updating, its pointing to version 11

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one other very small comment. Thanks

File: README.md (hunk 1 above)
Comment (MancunianSam) [reply README.md]: Ah yeah, I've updated it to 18.0. They don't seem to do a `latest` url so we'll have to keep this up to date next time we do a major version upgrade.

**Pull request merged**